mite_selenium/__init__.py

```diff
@@ -48,45 +52,145 @@ def start(self):
+        else:
+            logger.info("Secure TLS connection not used, defaulting tls_time to 0")
+            return 0
+
+    def _get_tcp_timing(self, timings):
+        if self._is_using_tls(timings["name"]):
+            return timings["secureConnectionStart"] - timings["connectStart"]
+        else:
+            return timings["connectEnd"] - timings["connectStart"]
+
+    def _send_page_load_metrics(self):
+        if self._browser_has_timing_capabilities():
+            performance_entries = self._remote.execute_script(
+                "return performance.getEntriesByType('navigation')"
+            )
+
+            timings = self._extract_first_entry(performance_entries)
+            if timings is None:
+                return
+
+            protocol = timings["nextHopProtocol"]
+            if protocol != "http/1.1":
+                logger.warning(
+                    f"Timings may be inaccurate as protocol is not http/1.1: {protocol}"
+                )
+            metrics = {
+                "dns_lookup_time": timings["domainLookupEnd"]
+                - timings["domainLookupStart"],
+                "dom_interactive": timings["domInteractive"],
+                "js_onload_time": timings["domContentLoadedEventEnd"]
+                - timings["domContentLoadedEventStart"],
+                "page_weight": timings["transferSize"],
+                "render_time": timings["domInteractive"] - timings["responseEnd"],
+                "tcp_time": self._get_tcp_timing(timings),
+                "time_to_first_byte": timings["responseStart"] - timings["connectEnd"],
+                "time_to_interactive": timings["domInteractive"]
+                - timings["requestStart"],
+                "time_to_last_byte": timings["responseEnd"] - timings["connectEnd"],
+                "tls_time": self._get_tls_timing(timings),
+                "total_time": timings["duration"],
+            }
+            self._context.send(
+                "http_selenium_page_load_metrics",
+                **self._extract_and_convert_metrics_to_seconds(metrics),
+            )
+
+    def _extract_first_entry(self, entries):
+        if len(entries) != 1:
+            logger.error(
+                f"Performance entries did not return the expected count: expected 1 - actual {len(entries)}"
+            )
+            return
+        else:
+            return entries[0]
+
+    def _extract_and_convert_metrics_to_seconds(self, metrics):
+        converted_metrics = dict()
+        non_time_based_metrics = ["page_weight"]
+        for k, v in metrics.items():
+            if k not in non_time_based_metrics:
+                converted_metrics[k] = self._convert_ms_to_seconds(v)
+            else:
+                converted_metrics[k] = v
+        return converted_metrics
+
+    def _convert_ms_to_seconds(self, value_ms):
+        try:
+            return value_ms / 1000
+        except Exception:
+            return value_ms
```

Comment:
Hmm.  What exceptions have you seen here?  This seems like an overly permissive code path.

Reply:
I think originally my thinking was around converting values that weren't meant to be converted to seconds, as I was lazily operating on all values. I had changed it since to not call the convert if nnot required so this can be simplified